I found out these tests are failing, please let me know if the same happened in the build or if it was something else:

```
============================== short test summary info =================================
FAILED tests/system/data_sources/test_sql_server.py::test_schema_validation_core_types_to_bigquery - AssertionError
FAILED tests/system/data_sources/test_sql_server.py::test_schema_validation_not_null_vs_nullable - AssertionError: Column: col_nn, status: fail != success
```

--- a/data_validation/schema_validation.py
+++ b/data_validation/schema_validation.py
@@ -237,8 +237,10 @@ def split_allow_list_str(allow_list_str: str) -> list:
     # I've not moved this patter to a compiled constant because it should only
     # happen once per command and I felt splitting the pattern into variables
     # aided readability.
-    nullable_pattern = r"!?"
-    precision_scale_pattern = r"(?:\((?:[0-9 ,\-]+|'UTC')\))?"
+    nullable_pattern = r"!?"  # Matches an optional '!'
+    # Matches precision/scale like (N), (N,M), ('UTC'), or (N, 'UTC')
+    # Crucially, [0-9 ,-]+ allows for digits, spaces, commas, and hyphens (for ranges like "1-18").
+    precision_scale_pattern = r"(?:\((?:[0-9 ,-]+(?:,[ ]*'UTC')?|'UTC')\))?"
     data_type_pattern = nullable_pattern + r"[a-z0-9 ]+" + precision_scale_pattern
     csv_split_pattern = data_type_pattern + r":" + data_type_pattern
     data_type_pairs = [

--- a/tests/system/data_sources/test_sql_server.py
+++ b/tests/system/data_sources/test_sql_server.py
@@ -293,7 +293,9 @@ def test_schema_validation_core_types_to_bigquery():
             # All SQL Server integers go to BigQuery INT64.
             "int8:int64,int16:int64,int32:int64,"
             # BigQuery does not have a float32 type.
-            "float32:float64"
+            "float32:float64,",
+            # SQL Server TIMESTAMP type has scale=7 on Ibis which does not happen in BigQuery.
+            "timestamp(7):timestamp,!timestamp(7):!timestamp,timestamp(7, 'UTC'):timestamp('UTC'),",
         ),
     )
 
@@ -312,6 +314,8 @@ def test_schema_validation_not_null_vs_nullable():
             "-sc=sql-conn",
             "-tc=bq-conn",
             "-tbls=pso_data_validator.dvt_null_not_null=pso_data_validator.dvt_null_not_null",
+            # SQL Server TIMESTAMP type has scale=7 on Ibis which does not happen in BigQuery.
+            "--allow-list=timestamp(7):timestamp,!timestamp(7):!timestamp,timestamp(7, 'UTC'):timestamp('UTC'),",
         ]
     )
     df = run_test_from_cli_args(args)

